pkg/util/util.go

```diff
@@ -91,3 +91,27 @@ func ValidateDriverName(driverName string) error {
 	}
 	return err
 }
+
+// GenerateVolID generates a volume ID based on passed in parameters and version, to be returned
+// to the CO system
+func GenerateVolID(monitors, id, key, pool, clusterID, objUUID string, volIDVersion uint16) (string, error) {
+	poolID, err := GetPoolID(monitors, id, key, pool)
+	if err != nil {
+		return "", err
+	}
+
+	// generate the volume ID to return to the CO system
+	vi := CSIIdentifier{
+		PoolID:          poolID,
+		EncodingVersion: volIDVersion,
+		ClusterID:       clusterID,
+		ObjectUUID:      objUUID,
+	}
+
+	volID, err := vi.ComposeCSIID()
+	if err != nil {
```

Comment:
as [ComposeCSIID](https://github.com/ceph/ceph-csi/blob/d02e50aa9b5af752ccdb53c79a1a14bf001af81c/pkg/util/volid.go#L77) in case of error returns the empty string and error  you can directly return it

Reply:
Done